contracts/allowlist/OperatorAllowlistUpgradeable.sol

```diff
@@ -0,0 +1,189 @@
+
+// Introspection
+import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+
+// Interfaces
+import {IOperatorAllowlist} from "./IOperatorAllowlist.sol";
+
+// Interface to retrieve the implemention stored inside the Proxy contract
+interface IProxy {
+    // Returns the current implementation address used by the proxy contract
+    function PROXY_getImplementation() external view returns (address);
+}
+
+/*
+    OperatorAllowlist is an implementation of a Allowlist registry, storing addresses and bytecode
+    which are allowed to be approved operators and execute transfers of interfacing token contracts (e.g. ERC721/ERC1155).
+    The registry will be a deployed contract that tokens may interface with and point to.
+    OperatorAllowlist is not designed to be upgradeable or extended.
+*/
+
+contract OperatorAllowlistUpgradeable is ERC165, AccessControlUpgradeable, UUPSUpgradeable, IOperatorAllowlist {
+    ///     =====   State Variables  =====
+
+    /// @notice Only REGISTRAR_ROLE can invoke white listing registration and removal
+    bytes32 public constant REGISTRAR_ROLE = bytes32("REGISTRAR_ROLE");
+
+    /// @notice Only UPGRADE_ROLE can upgrade the contract
+    bytes32 public constant UPGRADE_ROLE = bytes32("UPGRADE_ROLE");
+
+    /// @notice Mapping of Allowlisted addresses
+    mapping(address => bool) private addressAllowlist;
+
+    /// @notice Mapping of Allowlisted implementation addresses
+    mapping(address => bool) private addressImplementationAllowlist;
+
+    /// @notice Mapping of Allowlisted bytecodes
+    mapping(bytes32 => bool) private bytecodeAllowlist;
+
+    /// @notice storage gap for additional variables for upgrades
+    uint256[20] __gap;
+
+    ///     =====       Events       =====
+
+    /// @notice Emitted when a target address is added or removed from the Allowlist
+    event AddressAllowlistChanged(address indexed target, bool added);
+
+    /// @notice Emitted when a target smart contract wallet is added or removed from the Allowlist
+    event WalletAllowlistChanged(bytes32 indexed targetBytes, address indexed targetAddress, bool added);
+
+    ///     =====   Initializer  =====
+
+    /**
+     * @notice Grants `DEFAULT_ADMIN_ROLE` to the supplied `admin` address
+     * @param _roleAdmin the address to grant `DEFAULT_ADMIN_ROLE` to
+     * @param _upgradeAdmin the address to grant `UPGRADE_ROLE` to
+     */
+    function initialize(address _roleAdmin, address _upgradeAdmin) public initializer {
+        __UUPSUpgradeable_init();
+        __AccessControl_init();
+        _grantRole(DEFAULT_ADMIN_ROLE, _roleAdmin);
+        _grantRole(UPGRADE_ROLE, _upgradeAdmin);
+    }
+
+    ///     =====  External functions  =====
+
+    /**
+     * @notice Add a target address to Allowlist
+     * @param addressTargets the addresses to be added to the allowlist
+     */
+    function addAddressToAllowlist(address[] calldata addressTargets) external onlyRole(REGISTRAR_ROLE) {
```

Comment:
The function name should be addAddressToAllowLists.  That is with an s at the end to imply multiple addresses.

Reply:
would `addAddressesToAllowList` be clearer, as an alternative